src/Interpreters/executeQuery.cpp

```diff
@@ -355,6 +355,15 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
 {
     const auto current_time = std::chrono::system_clock::now();
 
+    auto & client_info = context->getClientInfo();
+
+    // If it's an initial query, set to current_time
```

Comment:
The reader will wonder - why it's only for non-internal queries.

PS. `If it's an initial query` -> `If it's not an initial query`.

Reply:
The comment indeed needs modification: `If it's not an internal query, and the initial query time is not yet set`

Internal queries are those executed without a client context, which should not have initial_query_start_time set.

It's possible to not have initial query time set for non-internal and non-initial queries. For example, the initiator is running an old version of clickhouse that don't have the feature yet.